src/mjolnir/valhalla_add_predicted_traffic.cc

```diff
@@ -61,7 +61,7 @@ struct TrafficSpeeds {
 };
 
 // Convert big endian bytes to little endian
-int16_t to_little_endian(const int16_t val) {
+int16_t to_little_endian(const uint16_t val) {
```

Comment:
To get rid of left shift in case of negative value. It also seems more logical if we treat the bytes to be swapped as something unsigned. It also helps to get rid of right shift of negative value which is implementation-defined (not UB but still...)